go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,338 @@
+//
+// Event with no properties description
+//
+// Owner: Test codegen.
+type EventNoProperties struct {
+	amplitude.Event
+}
+
+func (stronglyTypedEvent EventNoProperties) ToEvent() amplitude.Event {
+	return amplitude.Event{
+		EventType: "Event No Properties",
+	}
+}
+
+// EventObjectTypes
+//
+// [View in Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Event%20Object%20Types
+//
+// Event with Object and Object Array
+//
+// Owner: Test codegen
+//
+// Params:
+//	- RequiredObject: property object type
+//	- RequiredObjectArray: property object array type
+type EventObjectTypes struct {
+	RequiredObject      interface{}
+	RequiredObjectArray []interface{}
+}
+
+func (stronglyTypedEvent EventObjectTypes) ToEvent() amplitude.Event {
+	return amplitude.Event{
+		EventType: "Event Object Types",
+		EventProperties: map[string]interface{}{
+			"requiredObject":      stronglyTypedEvent.RequiredObject,
+			"requiredObjectArray": stronglyTypedEvent.RequiredObjectArray,
+		},
+	}
+}
+
+type Ampli struct {
+	Disabled bool
+	Client   amplitude.Client
+}
+
+// Load initializes the Ampli wrapper.
+// Call once when your application starts.
+func (a *Ampli) Load(options LoadOptions) {
+	a.Disabled = options.Disabled
+
+	if a.Client != nil {
+		log.Default().Printf("Warn: Ampli is already initialized. Ampli.load() should be called once at application start up.")
+
+		return
+	}
+
+	if options.Client.isEmpty() {
+		options.Client = LoadClientOptions{
+			Configuration: DefaultConfiguration,
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Client.APIKey or APIKey[options.Environment], " +
+			"or options.Client.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration.Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	return !a.Disabled
+}
+
+// Track tracks an StronglyTypedEvent.
+func (a *Ampli) Track(userID string, event StronglyTypedEvent, eventOptions amplitude.EventOptions) {
```

Comment:
We have `userId` (vs `userID`) in the other SDKs. Would be good to make it consistent.

Reply:
Can we use `ID` here? it is recommended by linter 
```
method parameter `userId` should be `userID` (golint)
```

Reply:
I noticed in go we are able to directly pass userId into identify groupIdentify, setGroup, and other events. It makes sense that the identify call sets the client userId, but I don't think we should set the client userId for other track events. 
Also why do we need to set the userId for setGroup?

Reply:
I think server side SDKs should accept userId since different events may belong to different users. But client side SDKs can ignore userId since all events are from the same user. Previous discussion: https://amplitude.slack.com/archives/C02E1H87E2C/p1660948525608159?thread_ts=1660947850.763319&cid=C02E1H87E2C